packages/admin-next/dashboard/src/v2-routes/api-key-management/api-key-management-list/components/api-key-management-list-table/use-api-key-management-table-query.tsx

```diff
@@ -0,0 +1,34 @@
+import { GetPublishableApiKeysParams } from "@medusajs/medusa"
+import { useQueryParams } from "../../../../../hooks/use-query-params"
+
+type UseApiKeyManagementTableQueryProps = {
+  prefix?: string
+  pageSize?: number
+}
+
+export const useApiKeyManagementTableQuery = ({
+  prefix,
+  pageSize = 20,
+}: UseApiKeyManagementTableQueryProps) => {
+  const queryObject = useQueryParams(
+    ["offset", "q", "created_at", "updated_at", "revoked_at", "order"],
+    prefix
+  )
+
+  const { offset, created_at, updated_at, revoked_at, q, order } = queryObject
+
+  const searchParams: GetPublishableApiKeysParams = {
+    limit: pageSize,
+    offset: offset ? Number(offset) : 0,
+    created_at: created_at ? JSON.parse(created_at) : undefined,
```

Comment:
Q: why do we need to parse the data here?

Reply:
Not sure, we've always done that. Maybe @kasperkristensen has the answer for this